esp/tools/soapplus/http.cpp

```diff
@@ -1229,40 +1268,130 @@ int HttpClient::sendStressRequest(StringBuffer& request, HttpStat* stat)
 
     unsigned start = msTick();
 
-    Owned<CSimpleSocket> sock = new CSimpleSocket(m_ssctx.get(), m_logfile);
-    
-    int ret = sock->connect(m_serveraddr.get());
-    if(ret < 0)
+    Owned<CSimpleSocket> sock;
+    bool isPersistent = m_globals->getPropBool("isPersist", false);
+    if (isPersistent)
     {
-        return -1;
+        if (persistentSocket.get() == nullptr)
+            persistentSocket.setown(new CSimpleSocket(m_ssctx.get(), m_logfile));
+        sock.set(persistentSocket.get());
+    }
+    else
+        sock.setown(new CSimpleSocket(m_ssctx.get(), m_logfile));
+    if (!sock->isConnected())
+    {
+        int ret = sock->connect(m_serveraddr.get());
+        if (ret < 0)
+        {
+            return -1;
+        }
     }
 
+    bool shouldClose = false;
     int sent = sock->send(request);
     __int64 total_len = 0;
-    if(sent >= 0)
+    StringBuffer xml;
+    if (sent > 0)
     {
-        int len = 1;
+        int len = 0;
         char recvbuf[2048];
-        StringBuffer xml;
-        while(1)
+        if (!isPersistent)
+        {
+            while (1)
+            {
+                len = sock->receive(recvbuf, 2047);
```

Comment:
This wont return now until either 2047 bytes are sent or the connection is closed.  Is that always ok ?